modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,691 @@
+                GAPI_Assert(false && "DAI outputs are not set");
+            }
+        }
+
+        // Properly link all nodes
+        // 1. Link input nodes to camera
+        for (const auto& d : ins_data)
+        {
+            for (const auto& nh : nodes)
+            {
+                if (m_gm.metadata(nh).contains<cv::gimpl::Op>())
+                {
+                    for (const auto& indata : nh.get()->inNodes()) {
+                        auto rc = m_gm.metadata(indata).get<cv::gimpl::Data>().rc;
+                        if (rc == d.rc)
+                        {
+                            GAPI_Assert(m_oak_nodes.find(nh) != m_oak_nodes.end());
+                            GAPI_Assert(m_oak_nodes[nh].inputs.size() == 1);
+                            // FIXME: covert other camera outputs
+                            m_camera_input->video.link(*(m_oak_nodes[nh].inputs[0]));
+                            m_oak_nodes[nh].processed = true;
+                        }
+                    }
+                }
+            }
+        }
+
+        // 2. Link output nodes to XLinkOut nodes
+        size_t out_counter = 0;
+        for (const auto& d : outs_data)
+        {
+            for (const auto& nh : nodes)
+            {
+                if (m_gm.metadata(nh).contains<cv::gimpl::Op>())
+                {
+                    for (const auto& outdata : nh.get()->outNodes()) {
+                        auto rc = m_gm.metadata(outdata).get<cv::gimpl::Data>().rc;
+                        if (rc == d.rc)
+                        {
+                            GAPI_Assert(m_oak_nodes.find(nh) != m_oak_nodes.end());
+                            GAPI_Assert(m_oak_nodes[nh].outputs.size() == 1);
+                            GAPI_Assert(out_counter < m_xlink_outputs.size());
+                            m_oak_nodes[nh].outputs[0]->link(m_xlink_outputs[out_counter++]->input);
+
+                            if (!m_oak_nodes[nh].processed) {
+                                LinkToParentHelper(nh, nodes);
+                                m_oak_nodes[nh].processed = true;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // 3. Link internal nodes to their parents
+        for (const auto& nh : nodes) {
+            if (m_gm.metadata(nh).get<NodeType>().t == NodeType::OP) {
+                GAPI_Assert(m_oak_nodes.find(nh) != m_oak_nodes.end());
+                if (!m_oak_nodes[nh].processed) {
+                    LinkToParentHelper(nh, nodes);
+                    m_oak_nodes[nh].processed = true;
+                }
+            }
+        }
+
+        m_device = std::unique_ptr<dai::Device>(new dai::Device(*m_pipeline));
+
+        // Prepare all output queues
+        for (size_t i = 0; i < outs_data.size(); ++i) {
+            // FIXME: add queue parameters
+            m_out_queues.push_back(m_device->getOutputQueue(m_out_queue_names[i], 30, true));
```

Comment:
It definitely need comments on that is this 30 number at least (I can guess that it's the fps, but still)

Reply:
It's not fps but a queue max capacity, added a comment